models/image.py

```diff
@@ -813,6 +883,54 @@ def from_new_and_ref(cls, new_image, ref_image):
         # Note that "data" is not filled by this method, also the provenance is empty!
         return output
 
+    def align_images(self):
+        """Align the upstream_images to one of the images pointed to by image_index.
+
+        The parameters of the alignment must be given in the parameters attribute
+        of this Image's Provenance.
+
+        The index to which the images are aligned is given by the parameters, using
+        the "to_index" key, which can be "first" or "last".
+
+        The resulting images are saved in _aligned_images, which are not saved
+        to the database. Note that each aligned image is also referred to by
+        a global variable under the ImageAligner.temp_images list.
+        """
+        from pipeline.alignment import ImageAligner  # avoid circular import
```

Comment:
If the point of this PR is to move image alignment out of the pipeline and into the image model, then the ImageAligner probably really belongs in the util subdirectory rather than the pipeline directory.

(I suspect we have other cases of things in model importing things in pipeline.)

This is the sort of thing that could go to a new Issue rather than being modified in this PR.